crates/risc0-bonsai/src/host.rs

```diff
@@ -316,13 +346,55 @@ impl<'a> ZkvmHost for Risc0BonsaiHost<'a> {
 
             // Start a session running the prover
             // execute only is set to false because we run bonsai only when proving
-            let session = retry_backoff_bonsai!(client.create_session(
-                hex::encode(self.image_id),
-                input_id.clone(),
-                vec![],
-                false
-            ))
-            .expect("Failed to create session; qed");
+            let client_clone = client.clone();
+            let image_id = hex::encode(self.image_id);
+            let input_id = input_id.clone();
+            let session = thread::spawn(move || {
```

Comment:
inlined these macros because I couldn't use the clone trick on Strings, open to suggestions